### DISCUSSION TITLE
Guide: Running beatoraja with Intel HD Graphics 2000 / 3000

### DISCUSSION THREAD

**Discussion by wcko87** [wcko87/beatoraja-english-guide#19] [beatoraja questions / troubleshooting]
Wrote this up a little while back. Gonna leave this here until I find a place to put it in the guide.

# tl;dr

beatoraja using the latest version of Java (8u60 and above) will not run with Intel HD Graphics 2000 and 3000.

We can however run beatoraja on these graphics processors if we use an older version of Java (8u51 and below). This is the same fix as used for Minecraft.

----
## A little more explanation

This is because Intel HD Graphics 2000 and 3000 are not supported on Windows 10.

These graphics processors technically support up to OpenGL 3.1, but they are not able to run OpenGL 3.1 on Windows 10 because of a small bug. And this prevents us from running beatoraja.

Intel refuses to update the drivers to support Windows 10 because the processors are marked "End of Life".
- https://www.intel.com/content/www/us/en/support/articles/000005524/graphics-drivers.html?eu-cookie-notice
- https://software.intel.com/en-us/forums/graphics-driver-bug-reporting/topic/607695

----
## What does this have to do with Java?

Java 8u60 and above recognize Windows 10 as an OS.
- https://github.com/LWJGL/lwjgl/issues/119#issuecomment-173983180 (scroll up slightly for details)

Java 8u51 and below do not. Which is why Intel HD Graphics 2000/3000 can still use OpenGL 3.1 properly on Java 8u51 and below.

----
# Solutions

## Solution 1:

Download a standalone version of JRE 8u51 and use that to run (beatoraja).
- https://github.com/LWJGL/lwjgl/issues/119#issuecomment-246253705

Note: This is an old version of Java, and may have security vulnerabilities. I suggest not using it for anything but running beatoraja.

In beatoraja_config.bat, you can replace "java" with the path to this java installation "blahblah\jre1.8.0_51\bin\java.exe"

## Solution 2:

Apply this patch by Marcinosoft:
- https://github.com/LWJGL/lwjgl/issues/119#issuecomment-264291065

This patch is a generic fix for Intel HD Graphics 2000/3000's bug.

Note: Because this patcher uses diablo2oo2's Universal Patcher 2.26.1 which is commonly used in cracks, the patcher will be flagged as malware.

Further information about this patcher:
- https://github.com/LWJGL/lwjgl/issues/119#issuecomment-262090230
- https://github.com/LWJGL/lwjgl/issues/119#issuecomment-262363814
